models/avatars/avatar.go

```diff
@@ -110,39 +104,73 @@ func HashedAvatarLink(email string, size int) string {
 		}
 		return lowerEmail, nil
 	})
-	if size > 0 {
-		return setting.AppSubURL + "/avatar/" + url.PathEscape(sum) + "?size=" + strconv.Itoa(size)
+	return emailHash
+}
+
+// GenerateUserAvatarFastLink returns a fast link (302) to the user's avatar: "/user/avatar/${User.Name}/${size}"
+func GenerateUserAvatarFastLink(userName string, size int) string {
+	if size < 0 {
+		size = 0
 	}
-	return setting.AppSubURL + "/avatar/" + url.PathEscape(sum)
+	//AppSubURL: It is either "" or starts with '/' and ends without '/', such as '/{subpath}'. question: should we use AppURL or StaticURLPrefix?
```

Comment:
You're pointing to /user/avatar/:username/:size here so AppSubURL is the correct thing.

```suggestion
```